src/keyring/KeyringController.ts

```diff
@@ -467,17 +465,12 @@ export class KeyringController extends BaseController<
       const privateKeyBuffer = toBuffer(addHexPrefix(privateKey));
       switch (version) {
         case SignTypedDataVersion.V1:
-          // signTypedDataLegacy will throw if the data is invalid.
-          return signTypedDataLegacy(privateKeyBuffer, {
-            data: messageParams.data as any,
-          });
         case SignTypedDataVersion.V3:
-          return signTypedData(privateKeyBuffer, {
-            data: JSON.parse(messageParams.data as string),
-          });
         case SignTypedDataVersion.V4:
-          return signTypedData_v4(privateKeyBuffer, {
-            data: JSON.parse(messageParams.data as string),
```

Comment:
These `JSON.parse` calls were removed, but we still need this to be JSON data rather than a string